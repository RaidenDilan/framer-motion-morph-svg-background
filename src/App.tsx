import React, { lazy, Suspense, useRef } from 'react';
import { Route, Routes } from 'react-router-dom';

import Layout from './hoc/Layout';
import { GlobalProvider } from './hooks/GlobalContext';

import { Spinner } from './components/Spinner';
import ErrorBoundary from './components/ErrorBoundary';

const Main = lazy(() => import('./screens/Main'));
const NotFound = lazy(() => import('./screens/NotFound'));

const App = () => {
  const nodeRef = useRef<HTMLDivElement>(null);
  return (
    <GlobalProvider>
      <div ref={nodeRef}>
        <Layout>
          <ErrorBoundary>
            <Suspense fallback={<Spinner />}>
              <Routes>
                <Route path="/" element={<Main />} />
                <Route path="*" element={<NotFound />} />
              </Routes>
            </Suspense>
          </ErrorBoundary>
        </Layout>
      </div>
    </GlobalProvider>
  );
};

export default App;
