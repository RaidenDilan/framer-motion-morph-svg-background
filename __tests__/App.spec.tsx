import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { act, render } from '@testing-library/react';
import { config } from 'react-transition-group';

config.disabled = true;

import App from '../src/App';

jest.mock('React', () => ({
  ...jest.requireActual('React'),
  useEffect: jest.fn(),
}));

describe('render the app', () => {
  it('should render without crashing', async () => {
    await act(async () => {
      const { asFragment } = render(
        <MemoryRouter>
          <App />
        </MemoryRouter>,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
