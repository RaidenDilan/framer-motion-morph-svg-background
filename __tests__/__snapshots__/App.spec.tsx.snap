// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the app should render without crashing 1`] = `<DocumentFragment />`;
